classes/mock/generator.php

```diff
@@ -805,77 +818,85 @@ protected static function generateUnknownClassCode($class, $mockNamespace, $mock
 
 	protected static function methodNameIsReservedWord(\reflectionMethod $method)
 	{
-		switch ($method->getName())
-		{
-			case '__halt_compiler':
-			case 'abstract':
-			case 'and':
-			case 'array':
-			case 'as':
-			case 'break':
-			case 'callable':
-			case 'case':
-			case 'catch':
-			case 'class':
-			case 'clone':
-			case 'const':
-			case 'continue':
-			case 'declare':
-			case 'default':
-			case 'die':
-			case 'do':
-			case 'echo':
-			case 'else':
-			case 'elseif':
-			case 'empty':
-			case 'enddeclare':
-			case 'endfor':
-			case 'endforeach':
-			case 'endif':
-			case 'endswitch':
-			case 'endwhile':
-			case 'eval':
-			case 'exit':
-			case 'extends':
-			case 'final':
-			case 'for':
-			case 'foreach':
-			case 'function':
-			case 'global':
-			case 'goto':
-			case 'if':
-			case 'implements':
-			case 'include':
-			case 'include_once':
-			case 'instanceof':
-			case 'insteadof':
-			case 'interface':
-			case 'isset':
-			case 'list':
-			case 'namespace':
-			case 'new':
-			case 'or':
-			case 'print':
-			case 'private':
-			case 'protected':
-			case 'public':
-			case 'require':
-			case 'require_once':
-			case 'return':
-			case 'static':
-			case 'switch':
-			case 'throw':
-			case 'trait':
-			case 'try':
-			case 'unset':
-			case 'use':
-			case 'var':
-			case 'while':
-			case 'xor':
-				return true;
-
-			default:
-				return false;
+		return in_array($method->getName(), self::getMethodNameReservedWordByVersion(), true);
+	}
+
+	protected static function getMethodNameReservedWordByVersion()
+	{
+		$methodNameReservedWord = array();
+
+		if(PHP_MAJOR_VERSION < 7) {
+			$methodNameReservedWord = array(
+				'abstract',
+				'and',
+				'array',
+				'as',
+				'break',
+				'callable',
+				'case',
+				'catch',
+				'class',
+				'clone',
+				'const',
+				'continue',
+				'declare',
+				'default',
+				'die',
+				'do',
+				'echo',
+				'else',
+				'elseif',
+				'empty',
+				'enddeclare',
+				'endfor',
+				'endforeach',
+				'endif',
+				'endswitch',
+				'endwhile',
+				'eval',
+				'exit',
+				'extends',
+				'final',
+				'for',
+				'foreach',
+				'function',
+				'global',
+				'goto',
+				'if',
+				'implements',
+				'include',
+				'include_once',
+				'instanceof',
+				'insteadof',
+				'interface',
+				'isset',
+				'list',
+				'namespace',
+				'new',
+				'or',
+				'print',
+				'private',
+				'protected',
+				'public',
+				'require',
+				'require_once',
+				'return',
+				'static',
+				'switch',
+				'throw',
+				'trait',
+				'try',
+				'unset',
+				'use',
+				'var',
+				'while',
+				'xor',
```

Comment:
I don't think this modification must be include in this PR. 

These keywords (yield and finally) are disallowed in 
@php >= 5.5 
@php < 7.0